comid/svn.go

```diff
@@ -182,6 +203,41 @@ func (o TaggedMinSVN) Valid() error {
 	return nil
 }
 
+// Compare compares SVN to a reference
+//
+// returns:
+//
+// 0 if they are equal
+// 1 if SVN is newer than the reference
+// -1 if SVN is older than reference or if the method encounters an error
+func (o TaggedMinSVN) Compare(r TaggedMinSVN) int {
```

Comment:
I'm wondering about the semantics of this.  I was expecting to compare a "normal" SVN against a "minSVN", no?  What does it mean to compare two "minSVN"s?

Reply:
That's true for verification, but in other contexts it might make sense? E.g. if you're comparing two policies. Also, changing the arument type would make the semantics for this `Compare` slightly different from the others (which compare like values, where as this compares against a threshold).

Maybe it would make sense to add a different method, `ValidateSVN(v TaggedSVN) bool` or something, for valiation (and a mirroring `CompareAgainstMinSVN(v TaggedMinSVN) bool` to `TaggedSVN`), and then either keep this method as-is or remove it we think that comparison between two MinSVN's never makes sense...

Reply:
> I'm wondering about the semantics of this. I was expecting to compare a "normal" SVN against a "minSVN", no? What does it mean to compare two "minSVN"s?

That's a good question, for Equality, I think they should exactly be the same.

But for comparison, I think the constraint on the reference doesn't matter. The minimum constraint only applies to the claim when comparing a MinSVN claim against a reference (ExactSVN or MinSVN).

For example, let's take the statement, "The minimum height to ride a roller coaster is 48 inches." There are two heights: the rider's and ruler's heights. When comparing, the minimum constraint only applies to the rider's height; the constraint on the ruler isn't something we care about for comparison (so it can be exact or minimal).

Reply:
>the constraint on the ruler isn't something we care about for comparison (so it can be exact or minimum).

In this example, a contraint on the ruler is something that doesn't really make sense, so we should not expose it as a method.